dev-python/sortedcontainers/sortedcontainers-2.1.0.ebuild

```diff
@@ -0,0 +1,22 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python2_7 python3_{6,7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="Python library to sort collections and containers"
+HOMEPAGE="http://www.grantjenks.com/docs/sortedcontainers/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+distutils_enable_tests pytest
+
+DEPEND="${PYTHON_DEPS}
+	dev-python/setuptools[${PYTHON_USEDEP}]"
+
+RDEPEND="${DEPEND}"
```

Comment:
I'd expect empty…
```
DEPEND=
RDEPEND=
```
…here and have the `distutils-r1` eclass pull in everything else — including setuptools and pytest — but I'll need to check again. 
For `PYTHON_DEPS` please drop and check https://wiki.gentoo.org/wiki/Project:Python/distutils-r1#PYTHON_DEPS .

Reply:
Are you sure about setuptools? I see it specified explicitly in a lot of packages in the dev-python category.

I'll drop the `PYTHON_DEPS` in the next commit

Reply:
Not 100% sure yet. I believe I saw something like it in the eclass. I'm running out of time for an in-depth look right now.

Reply:
I've been reading [this wiki page](https://wiki.gentoo.org/wiki/Project:Python/Dependencies#Reading_setuptools_setup.py_files) and if I understand it correctly both setuptools and `PYTHON_DEPS` should be specified:

> PYTHON_DEPS includes the dependency on the Python interpreter and dev-lang/python-exec. If the package installs Python scripts, modules or links to Python libraries, it needs to include PYTHON_DEPS in RDEPEND. In other words, every ebuild using distutils-r1, python-r1, or python-single-r1 needs to use it. 

> The easy rule is: always include PYTHON_DEPS in DEPEND. 

> Setuptools is an extension to the standard distutils build system commonly used by Python packages. In order to determine whether the package uses setuptools, you should check the setup.py file. If the setuptools module (or any function from it) is imported, then dev-python/setuptools needs to be specified as a build-time dependency.

Reply:
I like where this is going — interesting!

This is what empty `DEPEND` and `RDEPEND` in the ebuild get me with `distutils-r1`:
- `BDEPEND`: `test? ( pytest )` + `PYTHON_DEPS` + setuptools
- `DEPEND`: empty
- `RDEPEND`: `PYTHON_DEPS`

Unless `setuptools` is a true runtime dependency for this package — I find no imports to `setuptools` or `pkg_resources` outside of `setup.py` and no entrypoints in `setup.py` — that doesn't look too bad to me.

pytest was put in by `distutils_enable_tests pytest`:
https://github.com/gentoo/gentoo/blob/d51fec51c03a67074cf521d274e47ebcf1e1c412/eclass/distutils-r1.eclass#L396-L401

setuptools is in there because of default `${DISTUTILS_USE_SETUPTOOLS:=bdepend}` and its effect to `_distutils_set_globals`:
https://github.com/gentoo/gentoo/blob/cbb12d98d9e63b7e5f447aa170412a85550da5b4/eclass/distutils-r1.eclass#L118-L127

This is what I used as a quickhack to get at the final variable values:
```console
# ebuild sortedcontainers-2.1.0.ebuild manifest clean unpack
# for i in BDEPEND DEPEND RDEPEND ; do echo "[$i]"; ( source /var/tmp/portage/dev-python/sortedcontainers-2.1.0/temp/environment ; eval "echo \${$i}" ) | sed 's, ,\n,g' | sed 's,^,  ,' ; echo ; done
```

> The easy rule is: always include PYTHON_DEPS in DEPEND.

I think that's from a time before EAPI 7 was mainstream because `distutils-r1` today adds `PYTHON_DEPS` to [`BDEPEND`](https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html#build-time-dependencies-split-into-depend-and-bdepend) from EAPI 7 on (and to `DEPEND` before). To prove that with code:
https://github.com/gentoo/gentoo/blob/d51fec51c03a67074cf521d274e47ebcf1e1c412/eclass/distutils-r1.eclass#L134-L138

Since that wiki page is not specific to `distutils-r1`, maybe that advice was mean to be on the safe side even with ebuilds where only a small piece of the software uses Python — just guessing.
My source would be https://wiki.gentoo.org/wiki/Project:Python/distutils-r1#PYTHON_DEPS:
> [PYTHON_DEPS] should be used only with DISTUTILS_OPTIONAL enabled.


What do you think?